### ISSUE TITLE
`no-duplicate-selectors` false positives with SASS syntax

### ISSUE THREAD

**Issue by Katze-942** [stylelint/stylelint#7128]
### What minimal example or steps are needed to reproduce the bug?

```sass
body
  font:
    family: 'Arial', sans-serif
    weight: 400
    
.main
  width: 100vw
```

You need to create any .sass file with the code above.

### What minimal configuration is needed to reproduce the bug?

```json
{
  "extends": [
    "stylelint-config-standard-scss"
  ],
  "rules": {},
  "overrides": [
    {
      "files": ["*.sass"],
      "customSyntax": "postcss-sass"
    }
  ]
}
```

### How did you run Stylelint?

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIBG0CeAdAdgAmwMwkwBd4tdc8BDAWwEsAbVebAcgEEAnWy+lgGmwBnSpkEBaQTC54y5AO4xaAcwAWJbABYADFtm4sAOmqVamWXNpQiK5gEYdANzlYQfEHgYwAcjTiIYAB40AA70MAbCgoKu4IQeSgggwLLoIIFEMJhQgqnMANp62KmCRKhh9KZEYpCY8RJEIlCUHFASYFGpsgC6fCkgHACuYTkI2MAAvr04RSAQDlJcsCP5hcnT5DMew7nYBSAAVBGUHSA9hfrgAyUQ1ADKqMSUATupwRAl7eKRI4Xj3Vh-TAxGrxABiEA4xiIiQAVoJCDFYMFoog1hcSmUYBViC8QPRKBkSqkpujSuVKtU4sp6o1mlBcfjCURiX0MeTiJTatSSrSWm0TsxUoyYETXH03h8BTMAHoADgMGgMtgAnCzpq93kRPhJjssZVoDABWAy6EAAkDjIA)

### Which Stylelint-related dependencies are you using?

```json
{
  "stylelint": "latest",
  "stylelint-config-standard": "latest",
  "stylelint-config-standard-scss": "latest",
  "postcss": "^8.4.19",
  "postcss-sass": "^0.5.0"
}
```

### What did you expect to happen?

No mistake should have happened, because there are no duplicates.

### What actually happened?

Unexpected duplicate selector ".main ", first used at line 6 [(no-duplicate-selectors)](https://stylelint.io/user-guide/rules/no-duplicate-selectors)[6-7:1-16]

### Do you have a proposal to fix the bug?

I noticed that if you drop the body to the very bottom - the error suddenly disappears.
```sass
.main
  width: 100vw

body
  font:
    family: 'Arial', sans-serif
    weight: 400
```

The error also disappears if you get rid of font attachments, for example, by rewriting the code like this:
```sass
body
  font-family: 'Arial', sans-serif
  font-weight: 400
    
.main
  width: 100vw
```

**Comment by jeddy3** [comment]
@Katze-942 Thanks for the clear report, for using the template and for the demo.

I suspect this is an issue with the [`postcss-sass`](https://www.npmjs.com/package/postcss-sass) custom syntax rather than Stylelint itself. Sass, especially the whitespace version, is used less nowadays, and `postcss-sass` is unmaintained. It looks like someone forked it and published [`postcss-sass-2`](https://www.npmjs.com/package/postcss-sass-2), but that still carries the "Not all Sass syntax supported. Parser under development." warning. You can try that package, but it may still not support [declaration nesting](https://sass-lang.com/documentation/style-rules/declarations/#nesting).

Some options are:

- adopt vanilla CSS or non-whitespace SCSS ([as that parser seems to support declaration nesting](https://stylelint.io/demo/#N4Igxg9gJgpiBcIBG0CeACYAdAdu9AZhDgC7ya774ECGAtgJYA2q5A5AIIBODNTbAGnQBnGjmEBaYTB4EA3JSoB3GAwDmACzLoALAAY9CvOgC+uE+ly4AdHRoM82Y0oZQSG8gEYDANyVGzHBABEAJmGAA5ejhEGAAPegAHJhhrYTBhYWDwYjC1BBAnfCwQeJIYHChhEvIAbUVikGESVBSmBxIJSBw8qRIxKBouKCkM6pBFAF0BRRKuAFcU8fJgExnjEogfGR5YZfR643wiqkawpZqDkoAqNLGS6YaqErB55og6AGVUUho4y5KiQgzTGo0yJSegXwk3MuGy3TyADEIFw7CQCgArYTEbKwRJZRAnErNVowdqkAEgJg0crNErrRoktodLq5dR9AZDKCU6m0kj02ZNFrM0isnrs5qc4Zg-YlXkwOnBQVAkHghCWEAAPQAHNYdNZPABOAUbEAqkig9Jq8g3CE4EwgExAA))
- continue using whitespace Sass and avoid declaration nesting
- continue using whitespace Sass and declaration nesting and turn the rule off (although you may encounter other issues)

Adopting vanilla CSS and avoiding non-standard features like declaration nesting is the safest long-term bet.

**Comment by Katze-942** [comment]
Thank you very much for your reply. Yes, I noticed with SASS there are so many problems with many different linkers, so it is better to use SCSS because it is no different from SASS except syntax and also supports declaration nesting and other syntactic sugar.

With `postcss-sass-2` unfortunately nothing changes, the problems remain. I think it's worth waiting a while for `postcss-sass-2` to come out of development.

Probably should close the topic. 

> continue using whitespace Sass and declaration nesting and turn the rule off (although you may encounter other issues)

And yes, you're right. Disabling this rule seems to save the situation, but then there are strange anomalies that I can't even describe, when stylelint gives one error, then sometimes it miraculously disappears. And sometimes the problem occurs even in a trivial space at the end of a line. In general, there are a lot of problems, I realized that SASS is not for production.

**Comment by jeddy3** [comment]
@Katze-942 Thanks for the conclusion. I agree.

I think we can close the issue as:

- both syntaxes carry the `Not all Sass syntax supported` warning
- not an issue with Stylelint itself, rather a third party custom syntax
